server/src/test/java/org/elasticsearch/common/lucene/store/InputStreamIndexInputTests.java

```diff
@@ -243,6 +243,41 @@ public void testMarkRest() throws Exception {
         assertThat(is.read(), equalTo(2));
     }
 
+    public void testSkipBytes() throws Exception {
```

Comment:
This test looks to me like it would pass even without adding an implementation of `InputStreamIndexInput::skip`? Maybe we could wrap `input` in a spy to keep track of bytes skipped versus read at the `IndexInput` level and validate that against expectations?

Reply:
@bcully good idea, opened https://github.com/elastic/elasticsearch/pull/118485